jdi-light/src/main/java/com/epam/jdi/light/driver/WebDriverFactory.java

```diff
@@ -11,48 +11,82 @@
 import java.util.concurrent.locks.ReentrantLock;
 
-import static com.epam.jdi.light.common.Exceptions.*;
-import static com.epam.jdi.light.driver.get.DriverData.*;
-import static com.epam.jdi.light.driver.get.DriverTypes.*;
-import static com.epam.jdi.light.settings.JDISettings.*;
-import static com.epam.jdi.tools.StringUtils.*;
-import static com.epam.jdi.tools.map.MapArray.*;
-import static com.epam.jdi.tools.pairs.Pair.*;
+import static com.epam.jdi.light.common.Exceptions.exception;
+import static com.epam.jdi.light.driver.get.DriverData.DEFAULT_DRIVER;
+import static com.epam.jdi.light.driver.get.DriverTypes.CHROME;
+import static com.epam.jdi.light.driver.get.DriverTypes.getByName;
+import static com.epam.jdi.light.settings.JDISettings.getJDISettings;
+import static com.epam.jdi.tools.StringUtils.LINE_BREAK;
+import static com.epam.jdi.tools.map.MapArray.map;
+import static com.epam.jdi.tools.pairs.Pair.$;
 import static java.lang.String.format;
-import static java.lang.Thread.*;
-import static java.util.concurrent.TimeUnit.*;
-import static org.apache.commons.lang3.StringUtils.*;
+import static java.lang.Thread.currentThread;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
 
 /**
  * Created by Roman Iovlev on 26.09.2019
  * Email: roman.iovlev.jdi@gmail.com; Skype: roman.iovlev
  */
 public class WebDriverFactory {
-    private WebDriverFactory() { }
+    private static WebDriverFactory driverFactory;
 
-    public static MapArray<String, JFunc<WebDriver>> DRIVERS
-        = new MapArray<>(DEFAULT_DRIVER, () -> initDriver(CHROME));
-    public static boolean SINGLE_THREAD = false;
-    private static MapArray<String, WebDriver> RUN_DRIVERS = new MapArray<>();
-    private static Safe<MapArray<String, WebDriver>> THREAD_RUN_DRIVERS
-        = new Safe<>(MapArray::new);
+    private WebDriverFactory() {
+        RUN_DRIVERS = new MapArray<>();
+        THREAD_RUN_DRIVERS = new Safe<>(MapArray::new);
+        DRIVERS = new MapArray<>(DEFAULT_DRIVER, () -> initDriver(getByName(getJDISettings().DRIVER.name)));
+        SINGLE_THREAD = false;
+        INIT_THREAD_ID = -1;
+        SWITCH_THREAD = false;
+    }
 
-    public static boolean noRunDrivers() {
+    public static WebDriverFactory getWebDriverFactory() {
+        if (driverFactory == null) {
+            driverFactory = new WebDriverFactory();
+        }
+        return driverFactory;
+    }
+
+    public MapArray<String, JFunc<WebDriver>> getDrivers() {
+        return DRIVERS;
+    }
+
+    public long getInitThreadId() {
+        return INIT_THREAD_ID;
+    }
+
+    public void setInitThreadId(long Id) {
+        INIT_THREAD_ID = Id;
+    }
+
+    private final MapArray<String, JFunc<WebDriver>> DRIVERS;
+    private final boolean SINGLE_THREAD;
+    private MapArray<String, WebDriver> RUN_DRIVERS;
+    private final Safe<MapArray<String, WebDriver>> THREAD_RUN_DRIVERS;
+
+    private long INIT_THREAD_ID;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5482899)